monkey/infection_monkey/post_breach/pba.py

```diff
@@ -63,7 +63,10 @@ def run(self):
             result = exec_funct()
             if self.scripts_were_used_successfully(result):
                 T1064Telem(ScanStatus.USED, "Scripts were used to execute %s post breach action." % self.name).send()
-            PostBreachTelem(self, result).send()
+            if return_result:
+                return result
+            else:
+                PostBreachTelem(self, result).send()
```

Comment:
I don't like this change, method should do one thing only, and now it does two entirely different things based on a parameter. If you don't need to send telemetries, why can't you just execute the commands?

Reply:
Ah okay, I needed to store the result of the commands and then send it as one telemetry, which is why I was returning it. I'll change it back to overriding `run()` in ModifyShellStartupFile.